### PR TITLE
feat: Move the cmdline_to_json filter plugin from distributedci/dci-ansible into the redhatci.ocp Ansible collection (dci-labs/ansible-collection-redhatci-ocp) so the collection is self-sufficient and does not depend on dci-ansible for this filter.

### PR DESCRIPTION
**Pull request by fredericlepied**
## Summary

Move the cmdline_to_json filter plugin from distributedci/dci-ansible into the redhatci.ocp Ansible collection (dci-labs/ansible-collection-redhatci-ocp) so the collection is self-sufficient and does not depend on dci-ansible for this filter.

## Changes

Created plugins/filter/cmdline_to_json.py (filter plugin ported from dci-ansible with Apache 2.0 header, DOCUMENTATION/EXAMPLES/RETURN blocks, __future__ imports), created tests/unit/filter/test_cmdline_to_json.py (14 unit tests covering key=value, bare flags, nested dot keys, comma-separated values, BOOT_IMAGE exception, empty input, full cmdline, FilterModule class), modified roles/node_info/tasks/main.yml line 53 (cmdline_to_json → redhatci.ocp.cmdline_to_json FQCN). All 14 tests pass. GPG-signed commit.

TestHints: no-check

## Release Notes

* **New Features**
  * Added `cmdline_to_json` filter plugin that converts Linux kernel command lines into JSON objects, enabling structured processing of kernel parameters.

* **Documentation**
  * Updated plugin documentation to include the new filter.

* **Tests**
  * Added comprehensive unit test coverage for the filter plugin, including edge cases and nested parameter handling.

### CODE REVIEW

**Comment by fredericlepied** [comment]
recheck

**Comment by fredericlepied** [comment]
retest

**Comment by fredericlepied** [comment]
retest

**Review by tonyskapunk — CHANGES_REQUESTED** [review]
Please increase the spec and galaxy versions, and update the README.

I don't see dependency from dci-ansible. Have we been lucky the use of the filter is not failing on us because we have that satisfied through a different dependency?

**Comment by fredericlepied** [comment]
> Please increase the spec and galaxy versions, and update the README.

done

> I don't see dependency from dci-ansible. Have we been lucky the use of the filter is not failing on us because we have that satisfied through a different dependency?

I suppose so as it has been used only in the agents.

File: ansible-collection-redhatci-ocp.spec [hunk 1]
```diff
@@ -3,7 +3,7 @@
 %global forgeurl https://github.com/%{org}/%{repo}
 
 Name:           %{repo}
-Version:        3.3.EPOCH
+Version:        3.4.EPOCH
```
[the remark this answers is not in the document]
Comment (fredericlepied) [reply ansible-collection-redhatci-ocp.spec:6]: no need that's managed by the automation and that's the expected formats.

**Comment by fredericlepied** [comment]
retest

**Review by tonyskapunk — APPROVED** [review]
LGTM
